frontend/document/pdfdocument.lua

```diff
@@ -103,10 +103,17 @@ function PdfDocument:comparePositions(pos1, pos2)
 end
 
 function PdfDocument:getPageTextBoxes(pageno)
-    local page = self._document:openPage(pageno)
-    local text = page:getPageText()
-    page:close()
-    return text
+    local hash = "textbox|"..self.file.."|"..pageno
```

Comment:
Is this also sufficient in case of zoom changes and the like?

Reply:
I'm talking about playing around with this a bit basically.

![Screenshot_20230517_183217](https://github.com/koreader/koreader/assets/202757/abac7fc5-438b-4faa-8f9b-cd1de0616ae3)